cli/cmd/stat.go

```diff
@@ -39,6 +40,47 @@ func newStatOptions() *statOptions {
 	}
 }
 
+// Validate that the options do not contain mutually exclusive flags.
+func (s *statOptions) validateConflictingFlags() error {
+	if s.toResource != "" && s.fromResource != "" {
+		return errors.New("--to and --from flags are mutually exclusive")
+	}
+
+	if s.toNamespace != "" && s.fromNamespace != "" {
+		return errors.New("--to-namespace and --from-namespace flags are mutually exclusive")
+	}
+
+	return nil
+}
+
+// Special validation to be run if the target resource type is "namespace".
+func (s *statOptions) validateNamespaceFlags() error {
+	if s.toNamespace != "" {
+		return errors.New("--to-namespace flag is incompatible with namespace resource type")
+	}
+
+	if s.fromNamespace != "" {
+		return errors.New("--from-namespace flag is incompatible with namespace resource type")
+	}
+
+	// Note: technically, this allows you to say `stat ns --namespace default`, but that
+	// seems like an edge case.
+	if s.namespace != "default" {
+		return errors.New("--namespace flag is incompatible with namespace resource type")
+	}
+
+	return nil
+}
+
+func resourceType(from string) (string, error) {
+	elems := strings.Split(from, "/")
+	if len(elems) > 0 {
+		return k8s.CanonicalResourceNameFromFriendlyName(elems[0])
+	} else {
+		return "", errors.New("empty array did not contain a resource type")
+	}
+}
```

Comment:
We already have this functionality for parsing resource types, so I'd rather not duplicate it. At other place in this file where we parse resources from the command line, we use [util.BuildResource](https://github.com/linkerd/linkerd2/blob/master/controller/api/util/api_utils.go#L208), which should work for this use case as well.

Reply:
I think initially, I wasn't sure if `util.BuildResource` would do what I wanted, but after taking a closer look at the code, I believe it does. Thanks!